src/index.ts

```diff
@@ -39,7 +39,12 @@ function install(app: App, options: VueGtmUseOptions = { id: '' }): void {
 
   // Handle vue-router if defined
   if (options.vueRouter) {
-    void initVueRouterGuard(app, options.vueRouter, options.ignoredViews, options.trackOnNextTick);
+    void initVueRouterGuard({
+      app,
+      vueRouter: options.vueRouter,
+      ignoredViews: options.ignoredViews,
+      trackOnNextTick: options.trackOnNextTick
+    });
```

Comment:
Are there benefits of doing this?
The params in the JSDoc below would be now much more complicated :thinking: (see lint warnings)
Also this is internally anyways, so I'm unsure if I would like to have this in this PR

Reply:
The original code have the `ignoredViews` as an optional parameter, but was not at the end of the argument list.
even if this is allow in js, but in most of other languages, optional parameter should place at the end of the argument list. By making the argument to a single object would be more "standard" to my opinion.

Personally, I'd like to pass the argument as a single object, whenever the function has more than more than two parameters. 

To my experience, the benefits would be:
* safer to refactor
* easier to understand
* prevent passing any parameters to the wrong position

but anyway to is my opinion, if you'd like to have this as positional arguments, I'd like to revert it back👍🏼

Reply:
I once had a bad experience with it, so I accessed every param and one of them were optional. Then I missed to pass it in the calling side and therefor it was `undefined`. When I had to pass it explicitly as a x-th argument, I haven't missed it.

Maybe it would be okay here to pass app as first argument and options as second and then just pick what its needed.
But even then its not so much explicit :slightly_frowning_face: 

Also the JSDoc get a bit more complex by doing this.

In addition I need to keep this in sync with the vue2-gtm branch

https://github.com/gtm-support/vue-gtm/blob/9c3703caaabc914e64b7e6d8f7c0351cca2fb19b/src/index.ts#L80-L85

All in all, I think I don't see the need here, but also understand your point of view.

Reply:
Since we are using typescript, so shouldn't worry about miss passing the required fields of the typed object.

If you'd like to keep it as positional, could we just revert it back, then make the `trackOnNextTick` as optional as well because it should be optional anyway, and if you don't do this, the type of `ignoredViews` would be `any` which is not what we want.

Reply:
It has the exact same typedef due to the `VueGtmUseOptions['trackOnNextTick']` definition which is always the same as `options.trackOnNextTick`

I think all you need to do is changing `ignoredViews` typedef to `ignoredViews?: string[] | ((name: string) => boolean);`

And then use

```ts
if (Array.isArray(ignoredViews)) {
  ignoredViews = ignoredViews.map((view) => view.toLowerCase());
}
```

and

```ts
if (
  typeof to.name !== 'string' ||
  (Array.isArray(ignoredViews) && ignoredViews.indexOf(to.name.toLowerCase()) !== -1) ||
  (typeof ignoredViews === 'function' && ignoredViews(to.name))
) {
  return;
}
```

Reply:
Got it! Should I revert the argument list as positional one as well?